### ISSUE TITLE
Trying to Scan a jar file

### ISSUE THREAD

**Issue by dmodoomsirius** [gradle/gradle#1339]
### Expected Behavior

The jar to be signed.

### Current Behavior

When i applied the signing plugin and i have set it to sign the shadowJar and the Jar and when i run jarsigner -verify Jar.jar it outputs 

```
%userprofile%\Desktop\coding\git\MC\github\oblivion\Launcher\output\bootstrap>"C:\Program Files\Java\jdk1.8.0_65\bin\jarsigner.exe" -verify launcher-bootstrap-2.1.0-SNAPSHOT-all.jar
jar is unsigned. (signatures missing or not parsable)
```

### Context

I am wanting to sign the jar. 

### Steps to Reproduce (for bugs)

I followed https://docs.gradle.org/current/userguide/signing_plugin.html

### Your Environment

 * Build scan URL: https://scans.gradle.com/s/cknb5vt2b3heo
* Build.gradle: https://gist.github.com/dmodoomsirius/59146366f687dc331333c442183979f7

**Comment by Vampire** [comment]
For what purpose do you want to sign the JAR? Using jarsigner to verify the signature is wrong, as the signing plugin does not do the JAR signing that jarsigner is about. It creates a `.asc` signature file alongside your signed file, just as the documenation you linked to describes. Use PGP or GPG to verify the signature.

**Comment by oehme** [comment]
I adjusted the title and tags to match the problem: This is a feature request to support the jarsigner tool.

**Comment by Vampire** [comment]
@oehme acutally it is not. dmod was at IRC in parallel. He just wanted to know how to verify the signature the signing plugin was producing and thought `jarsigner` is the right tool for this.

But a support for JarSigner as task would be nice anyway. :-)

**Comment by oehme** [comment]
I see, thanks for clarifying that @Vampire! :)

**Comment by trejkaz** [comment]
We want to produce signed jars but hooking it in programmatically seems awkward because signing the file in-place wrecks out of date checking and hooking all the dependent tasks up to the signed jar instead is complex.

**Comment by trejkaz** [comment]
@Vampire as far as I can tell, if you get an exception during the doLast and then run the task again, it thinks it's already up-to-date and skips doing the signing. There might be a complicated workaround where you set the file modified date to something really low as soon as the doLast block starts, but you're entering hack territory at that point, and even setting the modified date could fail.

**Comment by Vampire** [comment]
There you either are wrong or found a regression bug that you should report @trejkaz.
If an exception happens in `doFirst` or `doLast`, the task fails.
If the task fails, it is never be up-to-date in the next run.
Setting file modification dates should have no effect at all, as Gradle compares checksums of the files, not file modification dates.

**Comment by Vampire** [comment]
**You** don't use the checksum, Gradle does. Simplified Gradle takes checksums of the input files and output files before the task is run and compares them to the input files before the last run and the output files after the last run of that task. If either doesn't match, meaning the input files were changed or the output files were changed after the last run, the task is out of date and run again. If they are the same, the task is considered up to date. After the task run means after all `doLast` actions were executed.  If the task fails, e. g. because of an exception in the `doLast` action, the task is always considered out of date, no matter what the input and output files look like.

If you can provide a reproducible example where this behaves different, you should open a separate issue with that example, because then that is a bug most likely.

The only cases where a task was considered up to date wrongly that I have seen were if either the inputs and outputs were not described properly, so Gradle didn't consider all relevant files which is a bug in your build script or the task in question, or when the Gradle caches got confused somehow which I occasionally have seen but cannot reproduce reliably. For that latter case it is necessary to delete the `.gradle` folder in the root project. After that all tasks are out of date but then it can calculate it properly again.

**Comment by Vampire** [comment]
@oehme ping

**Comment by oehme** [comment]
> I thought it should re-run because last execution failed.

Depends if it was stored in the cache. In that case it'll just be retrieved from the cache instead. Otherwise yes it would rerun.

**Comment by Vampire** [comment]
I didn't enable caching in my play-project, so I'll open a separate ticket, or didn't you mean the Gradle cache that is disabled by default?

**Comment by oehme** [comment]
Yes, please open a separate issue with the sample project.

**Comment by Vampire** [comment]
I reported it as #8018
@trejkaz I'm not fully sure that this is what you meant, but at least it is very similar.

**Comment by Vampire** [comment]
This would still be nice to have

**Comment by ljacomet** [comment]
There is no planned integration for this in Gradle core.
It could be provided by a plugin though.
